File: SOA-API/src/Core/Domain/Entities/ErrorLog.cs
using Domain.Abstractions;

namespace Domain.Entities
{
    public class ErrorLog : BaseEntity
    {
        public DateTime OccurredAtUtc { get; set; } = DateTime.UtcNow;
        public string Level { get; set; } = "Error";
        public string Message { get; set; } = string.Empty;
        public string? MessageKey { get; set; }
        public string? ExceptionType { get; set; }
        public string? StackTrace { get; set; }
        public int? StatusCode { get; set; }
        public string? TraceId { get; set; }
        public string? LoggingEvent { get; set; }
        public string? MemberName { get; set; }
        public int? LineNumber { get; set; }
        public string? RequestMethod { get; set; }
        public string? RequestPath { get; set; }
        public string? QueryString { get; set; }
        public string? UserId { get; set; }
        public string? UserName { get; set; }
        public string? Source { get; set; }
        public string? Extra { get; set; }
    }
}

File: SOA-UI/src/models/error-log.ts
export interface ErrorLogSummary {
  id: string;
  occurredAtUtc: string;
  level: string;
  message: string;
  statusCode?: number | null;
  requestPath?: string | null;
  requestMethod?: string | null;
  userId?: string | null;
  userName?: string | null;
  traceId?: string | null;
}

export interface ErrorLogDetail extends ErrorLogSummary {
  messageKey?: string | null;
  exceptionType?: string | null;
  stackTrace?: string | null;
  loggingEvent?: string | null;
  memberName?: string | null;
  lineNumber?: number | null;
  queryString?: string | null;
  source?: string | null;
  extra?: string | null;
}

export interface ErrorLogQueryParams {
  page?: number;
  pageSize?: number;
  fromUtc?: string;
  toUtc?: string;
  level?: string;
  statusCode?: number;
  path?: string;
  userId?: string;
  traceId?: string;
}

export interface ErrorLogPagedResponse {
  items: ErrorLogSummary[];
  page: number;
  pageSize: number;
  totalCount: number;
}
